starlite/response/base.py

```diff
@@ -119,6 +125,40 @@ def __init__(
         )
         self.raw_headers: List[Tuple[bytes, bytes]] = []
 
+    @classmethod
+    @deprecated(
+        "1.48.0",
+        pending=True,
+        info="Set `type_encoders` on a higher layer "
+        "instead or pass `type_encoders` to the `Response` instance directly",
+    )
+    def serializer(cls, value: Any) -> Any:
+        """Transform non-natively supported types into supported types.
+
+        Should raise `TypeError` if a type cannot be transformed into a supported type
+        """
```

Comment:
Hey @provinzkraut - as this is meant to be a deprecation, shouldn't this still return a serialized value?

Reply:
Nope. This method isn’t actually being used anymore by the serializer, unless you override it in a subclass. Look at the `get_serializer` method.

This is done to provide an easier upgrade path for us, avoid passing around the default serializer, and a slight performance benefit.

Reply:
But `Response.serializer()` is a documented method, and this changes its behavior.

E.g., this worked fine in the previous version:

```python
def default_serializer(value: Any) -> Any:
    """Serialize `value`.
    Args:
        value: To be serialized.
    Returns:
        Serialized representation of `value`.
    """
    if isinstance(value, pgproto.UUID):
        return str(value)
    return Response[Any].serializer(value)
```
Not a big issue, as now that I've pinpointed what is wrong I'm updating to the layered type encoders (which is a much better pattern), just wanted to check that this isn't an over site.

Smth similar going on in `starlite-pg-redis-docker` app, which I'll update now in due course.

Reply:
Ah, you’re right. I actually hadn’t thought of this case. Guess we can just revert to calling `default_serializer` in `serializer` to re-enable this. 
I’ll fix this tomorrow!

Reply:
@peterschutt Should be fixed by #1018

Reply:
Thanks!

Reply:
Just needs a review so I can merge it :wink: